server/auth/gatekeeper.go

```diff
@@ -129,9 +129,9 @@ func getAuthHeader(md metadata.MD) string {
 func (s gatekeeper) getClients(ctx context.Context) (versioned.Interface, kubernetes.Interface, *types.Claims, error) {
 	md, _ := metadata.FromIncomingContext(ctx)
 	authorization := getAuthHeader(md)
-	mode, err := s.Modes.GetMode(authorization)
-	if err != nil {
-		return nil, nil, nil, status.Error(codes.InvalidArgument, err.Error())
+	mode, valid := s.Modes.GetMode(authorization)
+	if !valid {
+		return nil, nil, nil, status.Error(codes.Unauthenticated, "token not valid for requested mode")
```

Comment:
When `authorization` is empty (`""`), `GetMode` used to return an error, which caused this function to return an error with `InvalidArgument`. If, in reality, there is no `authorization` yet, then we want to redirect our users to `/login`. We only perform that redirect when we return an `Unauthenticated` error.